### PR TITLE
Add ColorAxis component

### LINKED ISSUE #230: ColorAxis Component
#### Bug or Feature Request?
Feature

#### Description
Since Highcharts 7.2.0 colorAxis can be referenced with id from Series:
https://api.highcharts.com/highcharts/series.line.colorAxis

So now it is possible to create ColorAxis component without hacks.

There is still one unreleased fix that is needed for it to work:
https://github.com/highcharts/highcharts/pull/11832
As it is already merged to master, it probably is released in the next Highcharts version.

This is just a heads up as it was mentioned previously and I accidentally found the code on Highcharts repository.

### PR DESCRIPTION
**Pull request by anajavi**
Adds ColorAxis which can be referenced from series with colorAxisId.

Closes #230

Usage:
```jsx
<ColorAxis id="myColorAxis" min={0} max={30}>
  <ColumnSeries .../>
</ColorAxis>
<ColumnSeries colorAxisId="myColorAxis" .../>
```

Potential problem: it seems that Highcharts always uses coloraxis in series even if it is not set to use it. I think it uses the first coloraxis created (index 0), but I haven't tested.

PR does not have tests yet.

**This component goes by without using useDelayOnce. I think we should change useChart, useAxis and useSeries also. The useDelayOnce is kind of brittle and I fear it might break completely when React concurrent mode lands.**

@whawker what do you think?

### CODE REVIEW

File: packages/react-jsx-highcharts/src/components/Series/Series.js [hunk 1]
```diff
@@ -46,10 +48,11 @@ const Series = memo(
     const providerValueRef = useRef(null);
 
     const axis = useAxis(axisId);
+    const colorAxis = useColorAxis(colorAxisId);
```
Comment (whawker) [inline packages/react-jsx-highcharts/src/components/Series/Series.js]: Should this be wrapped in a `if (colorAxisId)`?

**Comment by anajavi** [comment]
> Althout whilst React is becoming concurrent, Highcharts certainly isn't, and things need to be executed is a specific order, (series added only after axis is created etc) so I suspect it may cause issues once removed?

Element that references axis before Axis element will break:
```jsx
<Series axisId="myaxis" />
<XAxis id="myaxis" />
```
But I see no reason to why it could not always be written in different order:
```jsx
<XAxis id="myaxis" />
<Series axisId="myaxis" />
```

Currently the useDelayOnce depends on axis create effect to have been ran. In the future the effect might actually run after the delay has passed.

Anyway we can bring the useDelayOnce easily back, if needed.

File: packages/react-jsx-highcharts/src/components/Series/Series.js (hunk 1 above)
Comment (anajavi) [reply packages/react-jsx-highcharts/src/components/Series/Series.js]: Hooks cannot be run conditionally. They must be run in the same order in every render.
https://reactjs.org/docs/hooks-rules.html

File: packages/react-jsx-highcharts/src/components/ColorAxis/ColorAxis.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        chart.needsRedraw();
+      }
+    };
+  }, []);
+
+  const modifiedProps = useModifiedProps(restProps);
+
+  useEffect(() => {
+    if (colorAxisRef.current !== null) {
+      if (modifiedProps !== false) {
```
[the remark this answers is not in the document]
Comment (anajavi) [reply packages/react-jsx-highcharts/src/components/ColorAxis/ColorAxis.js]: done

File: packages/react-jsx-highcharts/src/components/ColorAxis/ColorAxis.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    id: colorAxisId,
+    events,
+    ...nonEventProps
+  };
+};
+
+const createColorAxis = (chart, props) => {
+  const opts = getColorAxisConfig(props);
+  let colorAxis;
+  colorAxis = chart.addColorAxis(opts, false);
```
[the remark this answers is not in the document]
Comment (anajavi) [reply packages/react-jsx-highcharts/src/components/ColorAxis/ColorAxis.js]: That was leftover from previous code. Fixed.

**Comment by anajavi** [comment]
@whawker do you think this can now be merged or do we need more tests?

**Comment by whawker** [comment]
Looks good to me! Merging!

**Pull request merged**